mitodl/micromasters
# Added program selector menu

#### What are the relevant tickets?
Fixes #969 
Part of #796 

#### What's this PR do?
Adds a UI allowing the user to select their current program enrollment as well as some missing plumbing for the UI. This should allow the user to choose their current program and have this affect the search listings shown as well as the courses displayed in the dashboard. 

#### How should this be manually tested?
It is especially important to test the learner search manually since I couldn't get automated tests to work to assert this behavior. You should make sure that when you switch programs the search refreshes. If you go to the dashboard and back the search should also refresh. Basically you should never see one program's learners when the other program is selected from the dropdown.

If there is no program enrollment for a user (you can edit `getProgramEnrollments` to return a promise `[]` for this), you should see an error on both the learner and dashboard pages.

You should be able to open the dropdown and click anywhere on the page to close it again.

#### Any background context you want to provide?
Some infrastructure exists for the next PR which will add a dialog box to let the user select an enrollment

#### Screenshots (if appropriate)
![screenshot from 2016-09-08 11-33-17](https://cloud.githubusercontent.com/assets/863262/18355736/11bc7a1a-75b8-11e6-890b-988265996fea.png)
